projects/spectator/src/lib/matchers.ts

```diff
@@ -128,28 +189,74 @@ export const toHaveProperty = comparator((el, prop, val) => {
   return { pass, message };
 });
 
+export const toContainProperty = comparator((el, prop, val) => {
+  if (typeof prop === 'object') {
+    let pass = false;
+    let failing: string;
+
+    for (const key of Object.keys(prop)) {
```

Comment:
The only difference from `toHaveProperty` is the method name, right? You can create one function and pass either `containsProperty` or `hasProperty`.

Reply:
Sorry, I'm not sure I understand your suggestion. Can you explain it a little more?

Reply:
Something like:

```ts
function checkProperty(value, predicate) {
    let pass = false;
    let failing: string;

    for (const key of Object.keys(prop)) {
      const actual = $(el).prop(key);
      const addendum = prop[key] !== undefined ? ` with value '${prop[key]}'` : '';
      pass = predicate(actual, prop[key]);
      failing = !pass ? `'${prop}'${addendum}, but had '${actual}'` : '';
    }

    return { pass, failing };
}
```

`checkProperty(prop, containsProperty)`;
`checkProperty(prop, hasProperty)`;